node/packages/websocket/src/websocket.ts

```diff
@@ -0,0 +1,92 @@
+import { Logger } from '@shapeshiftoss/logger'
+import WebSocket from 'ws'
+
+const BASE_DELAY = 500
+const MAX_DELAY = 120_000
+const MAX_RETRY_ATTEMPTS = 0
+
+export interface Subscription {
+  jsonrpc: '2.0'
+  id: string
+  method: string
+  params?: unknown
+}
+
+export interface Args {
+  apiKey?: string
+  logger: Logger
+}
+
+export interface Options {
+  pingInterval?: number
+  retryAttempts?: number
+}
+
+export abstract class WebsocketClient {
```

Comment:
Contains all of the base connection and stay alive logic requiring `onOpen` and `onMessage` to be implemented.